config-provisioning/src/test/java/com/yahoo/config/provision/serialization/AllocatedHostsSerializerTest.java

```diff
@@ -72,6 +73,7 @@ private void assertAllocatedHosts(AllocatedHosts expectedHosts, NodeFlavors conf
             assertEquals(expectedHost.networkPorts(), deserializedHost.networkPorts());
             assertEquals(expectedHost.aliases(), deserializedHost.aliases());
             assertEquals(expectedHost.requestedResources(), deserializedHost.requestedResources());
+            assertEquals(expectedHost.dockerImageRepo().orElse(""), deserializedHost.dockerImageRepo().orElse(""));
```

Comment:
I think you can just compare the Optionals directly
```suggestion
            assertEquals(expectedHost.dockerImageRepo(), deserializedHost.dockerImageRepo());
```